### ISSUE TITLE
[Bug] User Data Context not available after refresh

### ISSUE THREAD

**Issue by chadcrotchett** [dev-launchers/strapiv4#218]
Per Dave (dbradham): We have observed that the user data context is not available after reloading the page. Specifically, we have a useEffect statement that executes when the page loads, and when it loads the user data context's user id, it is set to 0.

We came to the conclusion that this was out of the scope of the tickets being QA tested, and wanted to see if there is any ongoing efforts around the user data context.

**Describe the bug**
User Data is not available during a refresh of a page

**To Reproduce**
1. Login to staging environment when the vote button is enabled (confirm with Dave that it's enabled)
2. Select "Dream" then "Help Existing Idea"
3. Select a working idea (if it 404's then select another Idea): Working idea: https://staging.devlaunchers.org/ideaspace/workshop/259
4. Select "Upvote" button and button turns to "Upvoted" and number is incremented by 1
5. Refresh page
6. Observe that "Upvote" button does NOT say "Upvoted" and is not active - cannot vote or remove vote

**Expected behavior**
Upvote button still says "Upvoted" with total number of upvotes displayed

**Screenshots**
https://github.com/user-attachments/assets/b1a40d96-32ac-4945-8994-f2c3b23ac71d

**Environment**
 Staging

**Additional context**
Dave or Remi may have more details

**Comment by chadcrotchett** [comment]
@dbradham I've added a video. It seems though that the button is no longer read-only after a refresh as previously observed. Can you confirm the User Data Context timing is still the primary suspect?

**Comment by dbradham** [comment]
@chadcrotchett this is because we fixed the backend issue which only allowed 1 upvote per idea. The problem here is still the same as what we reported last Wednesday.

We are not able to determine if the user who is currently logged in has upvoted or not because we do not have the data for the user who is currently logged in after refreshing the page. There is a delay and the upvote button loads faster than we get user data context. I believe user data context is already being saved to local storage, so not sure why another request is sent to the backend when the page is refreshed.
